core/src/main/java/org/apache/accumulo/core/fate/accumulo/AccumuloStore.java

```diff
@@ -0,0 +1,288 @@
+      }
+    }
+
+    @Override
+    public long timeCreated() {
+      verifyReserved(false);
+
+      return scanTx(scanner -> {
+        scanner.setRange(getRow(tid));
+        TxColumnFamily.CREATE_TIME_COLUMN.fetch(scanner);
+        return StreamSupport.stream(scanner.spliterator(), false)
+            .map(e -> Long.parseLong(e.getValue().toString())).findFirst().orElse(0L);
+      });
+    }
+
+    @Override
+    public void push(Repo<T> repo) throws StackOverflowException {
+      verifyReserved(true);
+
+      try {
+
+        // TODO: should we push find top into the mutator?
+        // We also likely need a conditional mutation to make sure we are incrementing the latest
+        Optional<Integer> top = findTop();
+
+        if (top.filter(t -> t >= maxRepos).isPresent()) {
+          throw new StackOverflowException("Repo stack size too large");
+        }
+
+        FateMutator<T> fateMutator = newMutator(tid);
+        fateMutator.putRepo(top.map(t -> t + 1).orElse(1), repo).mutate();
+      } catch (StackOverflowException soe) {
+        throw soe;
+      }
+    }
+
+    @Override
+    public void pop() {
+      verifyReserved(true);
+
+      // TODO: should we push find top into the mutator?
+      // We also likely need a conditional mutation to make sure we are deleting top
+      Optional<Integer> top = findTop();
+      top.ifPresent(t -> {
+        newMutator(tid).deleteRepo(t).mutate();
+      });
+    }
+
+    @Override
+    public void setStatus(TStatus status) {
+      verifyReserved(true);
+
+      newMutator(tid).putStatus(status).mutate();
+
+      observedStatus = status;
+    }
+
+    @Override
+    public void setTransactionInfo(TxInfo txInfo, Serializable so) {
+      verifyReserved(true);
+      FateMutator<T> fateMutator = newMutator(tid);
+      final byte[] serialized = serializeTxInfo(so);
+
+      switch (txInfo) {
+        case TX_NAME:
+          fateMutator.putName(serialized);
+          break;
+        case AUTO_CLEAN:
+          fateMutator.putAutoClean(serialized);
+          break;
+        case EXCEPTION:
+          fateMutator.putException(serialized);
+          break;
+        case RETURN_VALUE:
+          fateMutator.putReturnValue(serialized);
+          break;
+        default:
+          throw new IllegalArgumentException("Unexpected TxInfo type " + txInfo);
+      }
+
+      fateMutator.mutate();
+    }
+
+    @Override
+    public void delete() {
+      verifyReserved(true);
+
+      newMutator(tid).delete().mutate();
+    }
+
+    private Optional<Integer> findTop() {
+      return scanTx(scanner -> {
+        scanner.setRange(getRow(tid));
+        scanner.fetchColumnFamily(RepoColumnFamily.NAME);
+        return StreamSupport.stream(scanner.spliterator(), false)
+            .map(e -> restoreRepo(e.getKey().getColumnQualifier())).findFirst();
+      });
+    }
+  }
+
+  static Text invertRepo(int position) {
+    return new Text(String.format("%02d", maxRepos - position));
```

Comment:
Could add an arg check to validate the passed in position.  Seems like it needs be in the range [1,100]?

Reply:
yeah i can add some validation checks to make sure the current values are in range. I just kept it to 100 for now but we can change it if needed since its arbitrary